core/Tracker/Response.php

```diff
@@ -161,6 +172,40 @@ private function outputTransparentGif()
         echo base64_decode($transGifBase64);
     }
 
+    /**
+     * Output a custom tracking image
+     *
+     * @param string $customImage The custom image setting specified in the config
+     * @return bool True if the custom image was successfully output, else false
+     */
+    private function outputCustomImage(string $customImage): bool
+    {
+
+        if (strlen($customImage) > 2 && substr($customImage, -2) == '==') {
+
+            $transGifBase64 = $customImage;
+            Common::sendHeader('Content-Type: image/gif');
+
+            echo base64_decode($transGifBase64);
+            return true;
+        }
+
+        if (file_exists($customImage)) {
+
+            $fp = fopen($customImage, 'rb');
+            $size = getimagesize($customImage); // imagesize is used to get the mime type
+            if ($fp && $size && isset($size['mime']))
+            {
+                Common::sendHeader('Content-Type: '.$size['mime']);
```

Comment:
Was wondering if we maybe need to escape the mime type from a security perspective but probably not needed I think and could potentially maybe break behaviour.

Reply:
Taking a quick look at the getimagesize php c source, it seems to be matching known byte headers and then returning a predefined mime type, so it doesn't look like there would be any way to usea carefully crafted image file to inject strings into the getimagesize return array. For extra safety we could check that the $size['mime'] string always matches a limited list of mime type we want to support, eg. image/gif, jpg or png.

Reply:
Mime types are now limited to restricted list